File: app/src/main/java/com/zhoujiulong/simpleprojectframework/base/BaseActivity.java
package com.zhoujiulong.simpleprojectframework.base;

import com.zhoujiulong.baselib.base.SimpleActivity;
import com.zhoujiulong.widgetlib.dialog.LoadingDialog;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class BaseActivity<T extends BasePresenter> extends SimpleActivity implements BaseView {

    /**
     * 網絡請求標記tag
     */
    protected String ReTag = String.valueOf(System.currentTimeMillis());
    /**
     * 用于添加rx的监听的在onDestroy中记得关闭不然会内存泄漏。
     */
    private CompositeDisposable mCompositeDisposable;

    protected T mPresenter;
    protected LoadingDialog mLoadingDialog;

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        if (this.mCompositeDisposable != null && !mCompositeDisposable.isDisposed()) {
            this.mCompositeDisposable.dispose();
        }
        if (mLoadingDialog != null && mLoadingDialog.isShowing()) {
            mLoadingDialog.dismiss();
        }
        mLoadingDialog = null;
        super.onDestroy();
    }

    @Override
    public void attachView() {
        if (mPresenter != null) {
            mPresenter.attachView(this, ReTag);
        }
    }

    @Override
    public void detachView() {
        if (mPresenter != null) {
            mPresenter.detachView();
            mPresenter = null;
        }
    }

    @Override
    public void bindSubscription(Disposable disposable) {
        if (this.mCompositeDisposable == null) {
            this.mCompositeDisposable = new CompositeDisposable();
        }
        this.mCompositeDisposable.add(disposable);
    }

    @Override
    public void showLoading() {
        if (mLoadingDialog == null) {
            if (mContext == null) return;
            mLoadingDialog = LoadingDialog.showLoading(mContext);
        } else {
            mLoadingDialog.show();
        }
    }

    @Override
    public void hideLoading() {
        if (mLoadingDialog != null) {
            mLoadingDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/zhoujiulong/simpleprojectframework/base/BaseActivity.kt
package com.zhoujiulong.simpleprojectframework.base

import com.zhoujiulong.baselib.base.SimpleActivity
import com.zhoujiulong.widgetlib.dialog.LoadingDialog
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

abstract class BaseActivity<T : BasePresenter<*, *>> : SimpleActivity(),
    BaseView {

    /**
     * 網絡請求標記tag
     */
    protected var ReTag = System.currentTimeMillis().toString()
    /**
     * 用于添加rx的监听的在onDestroy中记得关闭不然会内存泄漏。
     */
    private var mCompositeDisposable: CompositeDisposable? = null

    protected var mPresenter: T? = null
    private var mLoadingDialog: LoadingDialog? = null

    override fun onDestroy() {
        mCompositeDisposable?.apply { dispose() }
        mLoadingDialog?.apply { if (isShowing) dismiss() }
        super.onDestroy()
    }

    public override fun attachView() {
        mPresenter?.apply { attachView(this@BaseActivity, ReTag) }
    }

    public override fun detachView() {
        mPresenter?.apply { detachView() }
    }

    override fun bindSubscription(disposable: Disposable) {
        if (this.mCompositeDisposable == null) {
            this.mCompositeDisposable = CompositeDisposable()
        }
        this.mCompositeDisposable?.add(disposable)
    }

    override fun showLoading() {
        if (mLoadingDialog == null) mLoadingDialog = LoadingDialog.build(this)
        mLoadingDialog?.show()
    }

    override fun hideLoading() {
        mLoadingDialog?.dismiss()
    }
}









